### ISSUE TITLE
File Storage Delete

### ISSUE THREAD

**Issue by Ramu-kandimalla** [pucardotorg/dristi#1924]

**Comment by prasanna2400** [comment]
pr by subhashini :- https://github.com/pucardotorg/dristi-solutions/pull/566  , cc @manimaarans , @Ramu-kandimalla

**Comment by atulgupta2024** [comment]
@manimaarans include the DIGIT code and make required changes. Any code rabbit suggestions with regards code smells, refactoring etc. please ignore. We will fix any issues that we will have in the code we add. 

cc. @subhashini-egov @suresh12  FYI as well.

**Comment by eGovMan** [comment]
Why is delete required in file storage? Files are never deleted. New file is added and case can be pointed to the new file. 

You can define an archival strategy for files and data.

**Comment by atulgupta2024** [comment]
@eGovMan  this is for deletion of PDFs that are generated out of the case/applications etc and not any file that the advocate/ligitant my upload. As the case details are added, we have to generate a PDF and then e-sign it. The eSigned PDF is that is required to be saved. The initial PDF created is no longer required. 

Also after scrutiny if the case comes back to edit, the PDF gets generated again and then gets esigned again. 

Also due to draft application functionality, there are PDFs that get generated and overwritten as each new draft is saved. The earlier PDFs become orphaned and hence need to be deleted, else the system will get loaded with tons of orphaned PDFs overtime.

**Comment by subhashini-egov** [comment]
@Ramu-kandimalla this is still pending design review and sign off. I saw this is being picked up in the current sprint as "In progress".  I'd hold off on this until we finalize and instead pick some other ticket such as Security Audit log enablement.

**Comment by Susmitabe** [comment]
@subhashini-egov , please let us know if the design part is completed.

**Comment by Susmitabe** [comment]
Subhashini asked to keep it on hold

**Comment by atulgupta2024** [comment]
Multi party complaint e-sign, daily generation of case bundle, draft to published orders and various other scenarios will leave a trail of orphaned files. we will need to bring in **delete** functionality quickly, else we will end up with wasted storage space. 

cc. @suresh12 @subhashini-egov @anshumanth-egov

**Comment by Ramu-kandimalla** [comment]
Hi @atulgupta2024  @subhashini-egov  Please share the design here

**Comment by atulgupta2024** [comment]
@subhashini-egov please help. You were doing the design on this earlier.

**Comment by subhashini-egov** [comment]
@atulgupta2024 I need your help. Which services will need a delete API?

**Comment by atulgupta2024** [comment]
@subhashini-egov case, application, order for sure. Task service may also need it. but just like these use file store service to store a file and get back to store ID, will they just not call filestoresrv.delete(filestoreid) when required to delete a file?

**Comment by Ramu-kandimalla** [comment]
Hi @atulgupta2024 @subhashini-egov we are planning to close this in sprint 16 please let us know if the design part is  ready

**Comment by atulgupta2024** [comment]
@Ramu-kandimalla @manimaarans here's what needs to be done 

Extend the DIGIT file store service to include a delete API. It will be taking filestoreId as input. It will validate for existing of the document and then hard delete it. 

The respective entities like case, application, order can then call this API before they create and store the new PDF. _Will need to selectively called **only** where the earlier file has to be hard deleted._

**Comment by atulgupta2024** [comment]
@manimaarans correct. TenantId is passed in pretty much all APIs, so don't tend to list that separately. It is always assumed.

**Comment by bhuvanyuguru** [comment]
Working Fine in SOL-DEV 

![Image](https://github.com/user-attachments/assets/abf1abcf-64df-49e5-aec6-2860abd4d4b1)

**Comment by iknoorkaur** [comment]
Tested in sol-dev env, working as expected.
1. Currently, we have not implemented the delete filestore ID for evidence . Those attached documents will be linked to evidence.
2. Not deleting filestore id for complainant's and PoA Holder id proof.
3. Not deleting signed filestore id, that was generated before edit case as we are not storing it in documents. However, adding this filestore may cause an issue in the view case PDF.
Attaching jam videos for ref:
E-filing: https://jam.dev/c/cdb2638c-2395-47d8-937e-983aa63a3a02
https://jam.dev/c/b63e350d-8cbf-4c51-9af6-18ca26706797
Submiting the case: https://jam.dev/c/5ae635b5-87f0-4bc5-8783-25c9a1a211d2
returning back the case from fso: https://jam.dev/c/74babcec-d09d-4138-b311-2f003024acbd
returning back the case from judge: https://jam.dev/c/b4290e59-80fa-4d33-b0d5-d69a93191f71
applications: https://jam.dev/c/77c48c35-1eff-46fe-87df-336d870957b8
bulk reschedule: https://jam.dev/c/2096400b-c5df-487c-847e-0ec947ec9345
orders & a-diary: https://jam.dev/c/af281373-11a5-40be-8c72-5054bb9b249e
Issues identified: 
https://github.com/pucardotorg/dristi/issues/3913
https://github.com/pucardotorg/dristi/issues/3915
https://github.com/pucardotorg/dristi/issues/3916
https://github.com/pucardotorg/dristi/issues/3917
https://github.com/pucardotorg/dristi/issues/3912
https://github.com/pucardotorg/dristi/issues/3914
https://github.com/pucardotorg/dristi/issues/3951
https://github.com/pucardotorg/dristi/issues/3952
https://github.com/pucardotorg/dristi/issues/3955
https://github.com/pucardotorg/dristi/issues/3993

**Comment by atulgupta2024** [comment]
@rajeshcherukumalli @Ramu-kandimalla if this is solved, let's close this ticket

**Comment by rajeshcherukumalli** [comment]
Hi @atulgupta2024 for case this change is not implemented and remaning all places we are implemented this change and updated in production also.
cc @Ramu-kandimalla
